### PR TITLE
fix(webpack): allow transpilation of .cjs files

### PR DESCRIPTION
**Pull request by gnuletik**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

When trying to transpile a file from `node_modules` through the `build.transpile` property, files ending with `.cjs` were previously ignored.

This PR fixes it.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/config/base.js [hunk 1]
```diff
@@ -329,7 +329,7 @@ export default class WebpackBaseConfig {
         ]
       },
       {
-        test: /\.m?jsx?$/i,
+        test: /\.[mc]?jsx?$/i,
```
Comment (danielroe) [inline packages/webpack/src/config/base.js:332]: While we're at it:
```suggestion
        test: /\.[mc]?[tj]sx?$/i,
```

**Review by danielroe — APPROVED** [review]

**Comment by gnuletik** [comment]
@danielroe thanks for the approval !

For anyone interested in this feature before it gets merged, you can extend your webpack config like this : 
```ts
// nuxt.config.ts
export default {
// ...
  build: {
    extend (config: any, { isDev, isLegacy }: any) {
      // https://github.com/nuxt/nuxt.js/pull/10240
      const babelConfig = {
        configFile: false,
        babelrc: false,
        cacheDirectory: true,
        plugins: [
          '@nuxtjs/composition-api/dist/babel-plugin'
        ],
        presets: [
          ['@nuxt/babel-preset-app/src/index.js', {
            corejs: { version: 3 },
            jsx: { compositionAPI: true }
          }]
        ],
        envName: 'client'
      }

      if (!isDev && isLegacy) {
        config.module.rules.push({
          test: /\.c?jsx?$/i,
          exclude: (file: string) => {
            file = file.split(/node_modules(.*)/)[1]

            // not exclude files outside node_modules
            if (!file) {
              return false
            }

            const cjsTranspile = [/defu/]

            // item in transpile can be string or regex object
            return !cjsTranspile.some(module => module.test(file))
          },
          loader: 'babel-loader',
          options: babelConfig
        })
      }
   }
// ...
}
```
You can edit `cjsTranspile` to update the list of modules to transpile.

**Comment by pi0** [comment]
Thank you so much for making this PR @gnuletik ❤️  I'm closing it in favor of #10340 which has updated tests.

**Comment by kyumoon** [comment]
@gnuletik thanks for your config  code
in my case, i had to change the test regex to ` test: /\.cjsx?$/i,`

**Pull request closed without merging**